### DISCUSSION TITLE
Social auth redirect not working with staging environment.

### DISCUSSION THREAD

**Discussion by hevar** [supabase/supabase#17075] [Contribute to Supabase]
Production: `app.mydomain.com`
Staging: `test-app.mydomain.com`

When I login on `test-app.mydomain.com` through google oauth, I get redirect back to `app.mydomain.com` instead of `test-app.mydomain.com`. 

## System information

- OS: N/A - doesn't matter
- Browser: N/A - doesn't matter
- @supabase/auth-helpers-nextjs": "^0.7.4"
- @supabase/auth-helpers-react": "^0.3.1"
- @supabase/auth-ui-react": "^0.3.5"
- @supabase/auth-ui-shared": "^0.1.3"
- @supabase/supabase-js": "^2.21.0"

- Version of Node.js: Deployed on vercel - v. 18.x

![supabase-staging-auth](https://github.com/supabase/supabase/assets/9890748/55d347b9-d559-43f0-854e-aef192163408)

**Comment by saltcod** [comment]
What stack are you using @hevar?  Might be a better way to do this now if you're using Next

**Reply by hevar** [reply]
@saltcod I am using nextjs @latest (pages dir).
